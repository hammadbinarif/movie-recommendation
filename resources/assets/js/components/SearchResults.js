import React, { Component } from 'react';
import ReactDOM from 'react-dom';
import Movie from './Movie';

export default class SearchResults extends Component
{
    constructor(props, context) {
        super(props, context);

        this.state     =  {
            appState:               this.props.appState,
        };
    }

    /**
     * This function is called when parent updates props. In this event we will update state
     * @param newProps
     */
    componentWillReceiveProps(newProps) {
        this.setState({appState: newProps.appState});
    }

    /**
     * Render function
     * @returns {XML}
     */
    render() {
        return (
            <div className="container">
                <div className="form-group" />
                { this.state.appState.isSearched &&
                <table className="table table-striped">
                    <thead>
                    <tr>
                        <th scope="col">Movies</th>
                        <th scope="col">Rating</th>
                        <th scope="col">Genres</th>
                        <th scope="col">Show Time</th>
                    </tr>
                    </thead>
                    <tbody>
                    {
                        this.state.appState.movies && this.state.appState.movies.length > 0 && this.state.appState.movies.map((movie, index) => {
                            return <Movie movie={movie} key={index}/>
                        })
                    }
                    {
                        this.state.appState.isLoading === false && this.state.appState.movies.length === 0 &&
                        <tr>
                            <td colSpan={4}>no movie recommendations</td>
                        </tr>
                    }
                    </tbody>
                </table>
                }
            </div>
        );
    }
}