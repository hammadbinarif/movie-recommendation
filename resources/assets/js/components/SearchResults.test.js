// SearchResults.test.js
import React from 'react';
import SearchResults from './SearchResults';
import renderer from 'react-test-renderer';

test('SearchResults Test', () => {

    let state      = {
        genre:      'Drama',
        showTime:   '',
        isLoading:  false,
        isSearched: true,
        movies:     [],
        status:     '',
    };

    let setAppState = () => {};

    const component = renderer.create(
        <SearchResults appState={state} setAppState={setAppState} />,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});