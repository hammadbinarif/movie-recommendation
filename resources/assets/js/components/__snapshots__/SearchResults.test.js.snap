// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResults Test 1`] = `
<div
  className="container"
>
  <div
    className="form-group"
  />
  <table
    className="table table-striped"
  >
    <thead>
      <tr>
        <th
          scope="col"
        >
          Movies
        </th>
        <th
          scope="col"
        >
          Rating
        </th>
        <th
          scope="col"
        >
          Genres
        </th>
        <th
          scope="col"
        >
          Show Time
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td
          colSpan={3}
        >
          no movie recommendations
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
